airgun/views/modulestreams.py

```diff
@@ -0,0 +1,84 @@
+
+from widgetastic.widget import (
+    Table,
+    Text,
+    TextInput,
+    View,
+)
+from widgetastic_patternfly import BreadCrumb, Button
+
+from airgun.views.common import (
+    BaseLoggedInView,
+    SatTab,
+    SatTable,
+)
+from airgun.widgets import Search
+
+
+class CustomSearch(Search):
+    search_field = TextInput(locator=".//input[@role='combobox']")
+    search_button = Button('Search')
+
+
+class ModuleStreamView(BaseLoggedInView):
+    """Main Module_Streams view"""
+    title = Text("//h2[contains(., 'Module Streams')]")
+    table = SatTable('.//table', column_widgets={'Name': Text("./a")})
+
+    search_box = CustomSearch()
+
+    def search(self, query):
+        """Perform search using search box on the page and return table
+        contents.
+
+        :param str query: search query to type into search field. E.g.
+            ``name = "bar"``.
+        :return: list of dicts representing table rows
+        :rtype: list
+        """
+        self.search_box.search(query)
+        return self.table.read()
+
+    @property
+    def is_displayed(self):
+        """The view is displayed when it's title exists"""
+        return self.browser.wait_for_element(
+            self.title, exception=False) is not None
+
+
+class ModuleStreamDetailsTable(Table):
+
+    @property
+    def headers(self):
```

Comment:
could you please provide screenshot for that table? That is too low level logic that we should definitely avoid

Reply:
here is the screenshot 
![module stream details](https://user-images.githubusercontent.com/3190629/50827612-f0419d00-1364-11e9-8289-1d657b7b7a0c.png)

Reply:
1 sec, not this one

Reply:
I thought about this one: https://github.com/SatelliteQE/airgun/blob/master/airgun/widgets.py#L1333, but I need to see DOM structure. I just find something in common between your case and:
```
Example::
            {
                'Packages': '1',
                'Package Groups': '0'
                'Status': 'OK'
                'Domain': 'domain_name'
            }
```

Reply:
Anyway, you can look briefly on that class and check if you can re-use any idea used there

Reply:
let me check that once

Reply:
Added SatUnEvenTable to solve this